oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/web/BearerTokenAuthenticationFilter.java

```diff
@@ -144,10 +139,13 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
 	 * Set the {@link BearerTokenResolver} to use. Defaults to
 	 * {@link DefaultBearerTokenResolver}.
 	 * @param bearerTokenResolver the {@code BearerTokenResolver} to use
+	 * @deprecated Instead, use {@link BearerTokenAuthenticationConverter} explicitly
+	 * @see BearerTokenAuthenticationConverter
 	 */
+	@Deprecated
 	public void setBearerTokenResolver(BearerTokenResolver bearerTokenResolver) {
 		Assert.notNull(bearerTokenResolver, "bearerTokenResolver cannot be null");
-		this.bearerTokenResolver = bearerTokenResolver;
+		this.authenticationConverter.setBearerTokenResolver(bearerTokenResolver);
```

Comment:
I'd recommend something like this:

```java
Assert.isTrue(this.authenticationConverter instanceof BearerTokenAuthenticationConverter, "bearerTokenResolver and authenticationConverter cannot both be customized in this filter. Since you've customized the authenticationConverter, please consider configuring the bearerTokenResolver there.");
((BearerTokenAuthenticationConverter) this.authenticationConverter).setBearerTokenResolver(bearerTokenResolver);
```